src/main/java/hudson/maven/reporters/SurefireArchiver.java

```diff
@@ -138,7 +138,7 @@ public boolean postExecute(MavenBuildProxy build, MavenProject pom, MojoInfo moj
                 fileSet = Iterables.filter(fileSet, new Predicate<File>() {
                     @Override
                     public boolean apply(File input) {
-                        return !parsedFiles.containsKey(input);
+                        return !parsedFiles.containsKey(input) || parsedFiles.get(input) < input.lastModified();
```

Comment:
The pasedFiles Map<File, Long> (no longer a Set) does indeed store the modification timestamp of any pre-existing file - that was the essence of this change. Since the files in question are test results I think it's rather unlikely that they would have anything than a current timestamp.